cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -313,17 +313,18 @@ DefinitionByReferenceNode definitionByReferenceNodeFromArgument(Expr argument) {
 UninitializedNode uninitializedNode(LocalVariable v) { result.getLocalVariable() = v }
 
 private module ThisFlow {
-  private Node thisAccessNode(ControlFlowNode cfn) {
-    result.(ImplicitParameterNode).getFunction().getBlock() = cfn or
-    result.asExpr().(ThisExpr) = cfn
-  }
-
   private int basicBlockThisIndex(BasicBlock b, Node thisNode) {
-    thisNode = thisAccessNode(b.getNode(result))
+    // The implicit `this` parameter node is given a very negative offset to
+    // make space for any `ConstructorFieldInit`s there may be between it and
+    // the block contents.
+    thisNode.(ImplicitParameterNode).getFunction().getBlock() = b and
+    result = -1000
```

Comment:
What if there *are* 1000 `ConstructorFieldInit`s?

Reply:
I've pushed a commit to set this constant to the most negative integer allowed by QL. Surely that should be enough space for everyone.

Reply:
I'd prefer a cleaner solution, but this is good enough for me.